bin/check_dockerfiles.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+# Copyright 2019 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+BASE_DIR="$(cd "$(dirname "${0}")" && pwd -P)"
+ISTIO_ROOT="$(cd "$(dirname "${BASE_DIR}")" && pwd -P)"
+CD_TMPFILE=$(mktemp /tmp/check_dockerfile.XXXXXX)
+HL_TMPFILE=$(mktemp /tmp/hadolint.XXXXXX)
+
+find "${ISTIO_ROOT}" -name 'Dockerfile*' | \
+while read -r f; do
+  docker run --rm -i hadolint/hadolint < "$f" > "${HL_TMPFILE}"
```

Comment:
what would hadolint say about using an image with an explict tag :slightly_smiling_face: 

We may actually want to pin this though, otherwise if it updates and changes the lint logic our tests fail. This is especially bad for release branches -- when we went to make a change to release-1.0 all the lint logic had changed

Reply:
Good point. I'm going to pin to v1.17.1 which seems to be the latest version.